src/app/pages/profile/transaction-information/transaction-information.component.ts

```diff
@@ -0,0 +1,40 @@
+import { Component, OnInit } from '@angular/core';
+import { TransactionInformationService } from '../../../services/profile/transaction-information/transaction-information.service';
+import { APPCONFIG } from 'src/app/config';
+
+@Component({
+  selector: 'app-transaction-information',
+  templateUrl: './transaction-information.component.html',
+  styleUrls: ['./sass/transaction-information.component.scss']
+})
+export class TransactionInformationComponent implements OnInit {
+  base_url = APPCONFIG.base_url;
+  transacationUrl = `${this.base_url}/transactions/`;
+  sucessTransactions: any[] = [];
+  inProgressTransactions: any[] = [];
+  results: any[] = [];
+  resultSucces: any[] = [];
+  resultProgress: any[] = [];
+  constructor(private transactionService: TransactionInformationService) {}
+
+  ngOnInit(): void {
+    this.setTransacation(this.transacationUrl);
+  }
+  setTransacation(url: string) {
+    this.transactionService.getTransactions(url).subscribe(response => {
+      this.results = response.data.transactions;
+      if (this.results.length === 0) {
+        this.inProgressTransactions.length === 0;
```

Comment:
@Jaman-dedy , kindly explain what you want to accomplish inside this `if` block. The assignments expressions inside seem a bit redundant

Reply:
Thanks @Tevinthuku, I worked on refactoring it.

Reply:
I worked on that